### PR TITLE
[db] Implement new Team and Project entities 

### PR DESCRIPTION
**Pull request by jankeromnes**
## How to test

- (I'm not sure how to run the unit test)

- Log in to https://jx-teams-and-projects.staging.gitpod-dev.com/ and check the Console (it should log `Teams: []`)

- Connect to the DB, manually create a new team, and assign yourself to it:

```
mysql> insert into d_b_team (id, name, creationDate) values (UUID(), 'YOUR TEAM', '2021-06-03T10:00:00.000Z');

mysql> select * from d_b_team;

mysql> select id, name from d_b_user;

mysql> insert into d_b_team_membership (id, teamId, userId) values (UUID(), 'YOUR-TEAM-ID', 'YOUR-USER-ID');
```

- Refresh the dashboard -- the console should now log `Teams: [{ ... }]` (i.e. your team)

### CODE REVIEW

**Comment by jankeromnes** [comment]
Migration seems to work as expected:

<details>
<summary>(view MySQL tables)</summary>

```
mysql> describe d_b_project;
+-------------------+--------------+------+-----+----------------------+--------------------------------+
| Field             | Type         | Null | Key | Default              | Extra                          |
+-------------------+--------------+------+-----+----------------------+--------------------------------+
| id                | char(36)     | NO   | PRI | NULL                 |                                |
| cloneURL          | varchar(255) | NO   |     | NULL                 |                                |
| teamId            | char(36)     | NO   | MUL | NULL                 |                                |
| appInstallationId | varchar(255) | NO   |     | NULL                 |                                |
| connectedSince    | varchar(255) | NO   |     | NULL                 |                                |
| deleted           | tinyint(4)   | NO   |     | 0                    |                                |
| _lastModified     | timestamp(6) | NO   | MUL | CURRENT_TIMESTAMP(6) | on update CURRENT_TIMESTAMP(6) |
+-------------------+--------------+------+-----+----------------------+--------------------------------+
7 rows in set (0.00 sec)

mysql> describe d_b_team;
+---------------+--------------+------+-----+----------------------+--------------------------------+
| Field         | Type         | Null | Key | Default              | Extra                          |
+---------------+--------------+------+-----+----------------------+--------------------------------+
| id            | char(36)     | NO   | PRI | NULL                 |                                |
| creationDate  | varchar(255) | NO   |     | NULL                 |                                |
| deleted       | tinyint(4)   | NO   |     | 0                    |                                |
| _lastModified | timestamp(6) | NO   | MUL | CURRENT_TIMESTAMP(6) | on update CURRENT_TIMESTAMP(6) |
+---------------+--------------+------+-----+----------------------+--------------------------------+
4 rows in set (0.01 sec)

mysql> describe d_b_team_member;
+---------------+--------------+------+-----+----------------------+--------------------------------+
| Field         | Type         | Null | Key | Default              | Extra                          |
+---------------+--------------+------+-----+----------------------+--------------------------------+
| id            | char(36)     | NO   | PRI | NULL                 |                                |
| teamId        | char(36)     | NO   | MUL | NULL                 |                                |
| userId        | char(36)     | NO   | MUL | NULL                 |                                |
| deleted       | tinyint(4)   | NO   |     | 0                    |                                |
| _lastModified | timestamp(6) | NO   | MUL | CURRENT_TIMESTAMP(6) | on update CURRENT_TIMESTAMP(6) |
+---------------+--------------+------+-----+----------------------+--------------------------------+
```

</details>

Now I should probably add a few useful methods to the interfaces and verify they work.

File: components/gitpod-db/src/typeorm/entity/db-team-member.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { Entity, Column, PrimaryColumn, Index } from "typeorm";
+import { TypeORM } from "../typeorm";
+
+@Entity()
+// on DB but not Typeorm: @Index("ind_lastModified", ["_lastModified"])   // DBSync
+export class DBTeamMember {
```
Comment (svenefftinge) [inline components/gitpod-db/src/typeorm/entity/db-team-member.ts:12]: Can we call it Membership? Member seems to denote a person not the relation between team and user.

File: components/gitpod-db/src/typeorm/migration/1621920691254-InitialComSetup.ts [hunk 2]
```diff
@@ -1,27 +0,0 @@
-/**
```
Comment (svenefftinge) [inline components/gitpod-db/src/typeorm/migration/1621920691254-InitialComSetup.ts:1]: why was this removed?

Comment (jankeromnes) [reply components/gitpod-db/src/typeorm/migration/1621920691254-InitialComSetup.ts:1]: Please see the commit message:

>  [db] Remove unnecessary InitialComSetup migration (all tables already exist in Baseline)

This migration file is a no-op, because all tables it wants to create already existed before. @AlexTugarev and @geropl already agreed in principle. This is a minor drive-by fix, please ignore. 😇

File: components/gitpod-db/src/typeorm/entity/db-team-member.ts (hunk 1 above)
Comment (jankeromnes) [reply components/gitpod-db/src/typeorm/entity/db-team-member.ts:12]: Good point, thanks! Will adjust.

File: components/dashboard/src/App.tsx [hunk 3]
```diff
@@ -90,6 +90,13 @@ function App() {
         }
     }, []);
 
+    useEffect(() => {
+        (async () => {
+            const teams = await getGitpodService().server.getTeams();
+            console.log('Teams:', teams);
```
Comment (jankeromnes) [inline components/dashboard/src/App.tsx:96]: Note: This throws when not logged in -- a try/catch could be cleaner.

File: components/gitpod-db/src/team-db.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { Team } from "@gitpod/gitpod-protocol";
+
+export const TeamDB = Symbol('TeamDB');
+export interface TeamDB {
+    findTeamsByUser(userId: string): Promise<Team[]>;
```
Comment (jankeromnes) [inline components/gitpod-db/src/team-db.ts:11]: Potential improvements / follow-ups:
- create a new team
- delete a team
- add user to team
- remove user from team

File: components/dashboard/src/App.tsx (hunk 3 above)
Comment (svenefftinge) [reply components/dashboard/src/App.tsx:96]: Can we leave this test code out of the commit?

**Review by svenefftinge — APPROVED** [review]

File: components/gitpod-db/src/typeorm/entity/db-project.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { Entity, Column, PrimaryColumn, Index } from "typeorm";
+import { TypeORM } from "../typeorm";
+
+@Entity()
+// on DB but not Typeorm: @Index("ind_lastModified", ["_lastModified"])   // DBSync
```
[the remark this answers is not in the document]
Comment (jankeromnes) [reply components/gitpod-db/src/typeorm/entity/db-project.ts:11]: Yes, but it's true.

File: components/gitpod-db/src/typeorm/entity/db-project.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  @Column(TypeORM.UUID_COLUMN_TYPE)
+  @Index("ind_teamId")
+  teamId: string;
+
+  @Column()
+  appInstallationId: string;
+
+  @Column("varchar")
+  connectedSince: string;
```
[the remark this answers is not in the document]
Comment (jankeromnes) [reply components/gitpod-db/src/typeorm/entity/db-project.ts:27]: I agree. There is [a lot of variety](https://github.com/gitpod-io/gitpod/issues/4394) in the names and types of these fields, but `creationTime` does seem to be the more common name.

Will adjust. 👍

File: components/gitpod-db/src/typeorm/entity/db-team-membership.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Column(TypeORM.UUID_COLUMN_TYPE)
+  @Index("ind_teamId")
+  teamId: string;
+
+  @Column(TypeORM.UUID_COLUMN_TYPE)
+  @Index("ind_userId")
+  userId: string;
+
+  // This column triggers the db-sync deletion mechanism. It's not intended for public consumption.
```
[the remark this answers is not in the document]
Comment (jankeromnes) [reply components/gitpod-db/src/typeorm/entity/db-team-membership.ts:24]: Why add a `creationTime` to a team membership? (Note: There is already the `_lastModified` field, and I don't expect that we'll modify memberships -- just add or delete.)

File: components/gitpod-db/src/typeorm/entity/db-project.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/gitpod-db/src/typeorm/entity/db-project.ts:11]: ah, understood. thanks

File: components/gitpod-db/src/typeorm/entity/db-team-membership.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/gitpod-db/src/typeorm/entity/db-team-membership.ts:24]: I cannot access _lastmodified can I? In the member list I'd like to see `joined 2 months ago` or so.
